clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -318,6 +318,15 @@ static bool checkAttrMutualExclusion(Sema &S, Decl *D, const Attr &AL) {
   return false;
 }
 
+static bool checkDeprecatedFPGASYCLSpelling(Sema &S, const ParsedAttr &Attr) {
+  if (Attr.getScopeName()->isStr("intelfpga")) {
+    S.Diag(Attr.getLoc(), diag::warn_attribute_spelling_deprecated)
+        << "'" + Attr.getNormalizedFullName() + "'";
+    return true;
+  }
```

Comment:
We can make it lazier
```suggestion
  if (Attr.getScopeName()->isStr("intelfpga"))
    return S.Diag(Attr.getLoc(), diag::warn_attribute_spelling_deprecated)
        << "'" + Attr.getNormalizedFullName() + "'";
```

Reply:
Done